src/NanoMsg.cpp

```diff
@@ -0,0 +1,49 @@
+#include "NanoMsg.h"
+#include <cstring>
+#include <stdexcept>
+#include <g3log/g3log.hpp>
+/**
+ * Constructor for copying a packet hash vector into a nanomsg buffer
+ */
+NanoMsg::NanoMsg(PacketHashVector& data) : sent(false) {
```

Comment:
@john-gress, @yannick-polius : we can absolutely ignore std::vector to nanomsg copy. 

Instead of the std::vector the client can request a nanomsg of a certain size and then the constructor would allocate that buffer. 

The copy that right now happens is kind of there for compatibility with how it works today but since we know the copy is expensive it makes sense to not do it.